mass_mailing_partner/wizard/partner_merge.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2020 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class BasePartnerMergeAutomaticWizard(models.TransientModel):
+    _inherit = "base.partner.merge.automatic.wizard"
+
+    def mass_mailing_contact_items(self, partner_id):
+        return self.env['mail.mass_mailing.contact'].search([
+            ('partner_id', '=', partner_id)
+        ])
+
+    def _merge(self, partner_ids, dst_partner=None, extra_checks=True):
+        if dst_partner:
+            # search all records
+            all_contact_ids = self.env['mail.mass_mailing.contact'].search([
+                ('partner_id', 'in', partner_ids)
+            ])
+            if all_contact_ids:
+                item_contact_id = self.mass_mailing_contact_items(dst_partner.id)
+                # if record not exist (related to dst_partner) need to create
+                if not item_contact_id:
+                    self.env['mail.mass_mailing.contact'].create({
+                        'partner_id': dst_partner.id
+                    })
+                    item_contact_id = self.mass_mailing_contact_items(dst_partner.id)
+                # remove all records related to partner_ids
+                contact_ids = self.env['mail.mass_mailing.contact'].search([
+                    ('partner_id', 'in', partner_ids),
+                    ('partner_id', '!=', dst_partner.id)
+                ])
+                if contact_ids:
+                    list_ids = all_contact_ids.mapped('list_ids').ids
+                    list_ids = (
+                        list_ids + item_contact_id.list_ids.ids
+                    )
+                    contact_ids.unlink()
```

Comment:
You are missing here some info, like unsubscribe information.

Reply:
Thanks, in previous contact search filter to items with opt_out=False